### DISCUSSION TITLE
Delock (11827) Switching power is unreliable

### DISCUSSION THREAD

**Discussion by bobdig** [arendst/Tasmota#18006] [Q&A]
### PROBLEM DESCRIPTION
Switching power is unreliable for some time and versions now, no difference if physical button or WebUI is used. 
For example the problem is here while I am writing this report. 
I put in the commands like further down it is requested and now it is working again... Does this makes any sense?
Now that it is working again (don't know for how long) I can hear while switching power the click is louder. 

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Delock (11827)
- [x] Tasmota binary firmware version number used: 12.4.0 but problem is older
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
10:14:16.607 RSL: RESULT = {"NAME":"Delock 11827","GPIO":[0,0,0,32,2688,2656,0,0,2624,576,224,0,0,0],"FLAG":0,"BASE":53}
10:14:16.816 RSL: RESULT = {"Module":{"0":"Delock 11827"}}
10:14:17.022 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"32":"Button1"},"GPIO4":{"2688":"HLW8012 CF"},"GPIO5":{"2656":"HLWBL CF1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"2624":"HLWBL SEL_i"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"224":"Relay1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
10:15:34.496 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Delock_11827_006","FriendlyName":["Delock_11827_006"],"Topic":"tasmota_2E23CB","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
10:15:34.529 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:03:19","StartupUTC":"2023-02-20T09:12:15","Sleep":50,"CfgHolder":4617,"BootCount":176,"BCResetTime":"2021-08-11T10:58:03","SaveCount":1478,"SaveAddress":"FA000"}}
10:15:34.563 RSL: STATUS2 = {"StatusFWR":{"Version":"12.4.0(tasmota)","BuildDateTime":"2023-02-18T16:07:38","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"425/699"}}
10:15:34.583 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["guest2_optout_nomap",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A002800000000","00000081","00006000","00004000","00000000"]}}
10:15:34.615 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":632,"Free":368,"Heap":25,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
10:15:34.648 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-2E23CB-0971","IPAddress":"172.16.230.86","Gateway":"172.16.230.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.131.1","DNSServer2":"0.0.0.0","Mac":"F4:CF:A2:2E:23:CB","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
10:15:34.671 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_2E23CB","MqttUser":"tasmota","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
10:15:34.692 RSL: STATUS7 = {"StatusTIM":{"UTC":"2023-02-20T09:15:34","Local":"2023-02-20T10:15:34","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":99,"Sunrise":"07:49","Sunset":"18:17"}}
10:15:34.708 RSL: STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
10:15:34.727 RSL: STATUS10 = {"StatusSNS":{"Time":"2023-02-20T10:15:34","ENERGY":{"TotalStartTime":"2021-08-11T10:58:03","Total":13.464,"Yesterday":0.663,"Today":0.113,"Power":40,"ApparentPower":45,"ReactivePower":21,"Factor":0.89,"Voltage":234,"Current":0.194}}}
10:15:34.747 RSL: STATUS11 = {"StatusSTS":{"Time":"2023-02-20T10:15:34","Uptime":"0T00:03:19","UptimeSec":199,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":25,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"guest2_optout_nomap","BSSId":"BE:EE:7B:90:00:D4","Channel":13,"Mode":"11n","RSSI":100,"Signal":-49,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Now it is working again...


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
switching should work reliable. 

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Moving to discussions, asking for help. Nothing we can reproduce.

**Comment by barbudor** [comment]
Hi @bobdig
Your description 'switching is not reliable' is a bit vague
Can you please elaborate more?
How are you trigerring switching? Web UI? MQTT?
How can you show us the problem?
Thanks

**Comment by bobdig** [comment]
@barbudor This morning I noticed it again. On this plug I have a laptop. This laptop will start, when it gets "pluged in" (power from the grid). But the power-plug-LED on that laptop was already on, which shouldn't be the case, because there is a timer in the plug saying go off in the night. I then tried to turn off the plug by pushing its button. Only a light tone can be heard and it is actually not plugging anything. Tried it several times. I then opened that Laptop and pushing its power button. Now its starts up. I go to the webinterface and I am seeing in the console my switching, which hadn't any impact. 
But now when I switch the button, it does work again and it makes a somewhat louder sound!
It makes no sense.
Two things of note: The plugs had WiFi but no internet over the night time.
Plugged in is a laptop with a battery... 

And I don't use MQTT. 
I did several resets already, flashing minimal versions, using the template from [blakadder](https://templates.blakadder.com/delock_11827.html), nothing helps for a longer period but it does work some of the time.

**Comment by Jason2866** [comment]
When no NTP server is reachable Tasmota does not work.

**Comment by bobdig** [comment]
@Jason2866 So to my knowledge there was a NTP Server reachable, although I am not 100% sure. None the less, switching via Button or Webinterface should work in any case, at least in my mind.

**Reply by Jason2866** [reply]
Make sure WiFi is on and NTP Server is reachable. Everything else is guessing.

**Reply by bobdig** [reply]
> Make sure WiFi is on and NTP Server is reachable. Everything else is guessing.

It is kinda hard to test in my setup but I believe you now, that it is crucial to have NTP working. It is kinda sad though.

**Reply by barbudor** [reply]
I'm still unsure to understand what you mean by unreliable
Is that
- pressing the button on the plug doesn't always work?
   - does GUI button work then? 
- ghost power on or off while no one has pressed?

Regarding NTP, it often happens that internet routers can act as a local NTP server. If it isn't the router that is  switched off at night, you may use the NTPserver command to set it to the IP address of your router.

**Reply by bobdig** [reply]
> I'm still unsure to understand what you mean by unreliable Is that
> 
>     * pressing the button on the plug doesn't always work?
>       
>       * does GUI button work then?

No, both fail, they klick but don't switch power. Other things **"** work **"** like showing no power usage, what is actually wrong but seems hard coded to me. 

> Regarding NTP, it often happens that internet routers can act as a local NTP server. If it isn't the router that is switched off at night, you may use the NTPserver command to set it to the IP address of your router.

True but I have a router cascade and also blocking internet and or turn off a device sometimes so it is actually hard for me to test if NTP is working correctly for my IoT devices at night times. 

But I think it is a bug, that NTP is so crucial. My guess is that the the problem arises when there is a timer and NTP currently is not working. With that comes a state, after which Tasmota is not working reliably anymore, like described above. Manually switching (physical button or WebUI) should never be a problem in my book.

**Reply by barbudor** [reply]
Clicking but not actually switching seems to point in the direction of a failing device
I have a plug that fails in a similar way out of a pack of 3. Not the same brand nor firm factor as yours. As it's too complicated to open, I never tried to fix it.

**Reply by bobdig** [reply]
@barbudor I changed a firewall rule for NTP and I think it is working now all the time and I haven't had that problem for a few days now so I think @Jason2866 is correct. And that is a bug to me.

**Comment by bobdig** [comment]
I had the problem several times again, that the Delock plug is not switching power. I restarted the plug, didn't help, I turned off power for the plug itself, didn't helped. 
What helped was unplugging the laptop from the plug. 
So my guess now is that a device with its own battery can be a challenge here. Any thoughts?

**Reply by Jason2866** [reply]
No, since the switched power is fully decoupled via the Relais from the logic side.

**Reply by bobdig** [reply]
I don't think that the plug was actually powered by the laptop-battery. 
Maybe that battery itself can lead to switching problems for the plug but it is a wild guess, I don't know anything about power, relays, etc. I only know that unplugging it had helped.
